### PR TITLE
Refactored common java build bits into parent pom

### PR DESCRIPTION
**Pull request by bdemers**
* Adds new module with build related artifacts (pmd, license info, etc)
* Adds new module to build templated html to show multiple javadoc versions i.e. [this](https://developer.okta.com/okta-sdk-java/)
* handful of other items found while refactoring okta-sdk-java and related projects

### CODE REVIEW

File: doclist-plugin/src/main/java/com/okta/maven/doclist/DocListMojo.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2017 Okta, Inc.
```
[the remark this answers is not in the document]
Comment (bdemers) [reply doclist-plugin/src/main/java/com/okta/maven/doclist/DocListMojo.java]: +1 I fixed the date issue after I created that file

File: THIRD-PARTY-NOTICES [hunk 2]
```diff
@@ -13,10 +13,70 @@ See the License for the specific language governing permissions and
...
+  Plexus Security Dispatcher Component under Apache Public License 2.0
+  Plexus Velocity Component under Apache License, Version 2.0
+  Sisu - Guice under The Apache Software License, Version 2.0
+  Sisu - Inject (JSR330 bean support) under The Apache Software License, Version 2.0
+  Sisu - Inject (Plexus bean support) under The Apache Software License, Version 2.0
+  SLF4J API Module under MIT License
+  sslext under Apache Software License, Version 1.1
+  Struts Core under The Apache Software License, Version 2.0
+  Struts Taglib under The Apache Software License, Version 2.0
+  Struts Tiles under The Apache Software License, Version 2.0
```
Comment (mraible) [inline THIRD-PARTY-NOTICES]: Are you really using Struts in this project?

Comment (bdemers) [reply THIRD-PARTY-NOTICES]: ha, no, i need to check why that file didn't get updated.  I was thinking about writing a struts example (legacy migration use-cases), but stopped, i'll fix this

Comment (bdemers) [reply THIRD-PARTY-NOTICES]: wait... no... that isn't right i was thinking this was the `authn` repo. I'll check why that is getting pulled in.

**Pull request merged**